### PR TITLE
FLNK-1049 - Spec Validation using the JSON Schemas and Yup

### LINKED ISSUE #1049: Spec Validation using the JSON Schemas and Yup
Use the following library to convert JSON schemas to Yup schemas that we can use to validate schemas entered 
https://www.npmjs.com/package/json-schema-yup-transformer

### PR DESCRIPTION
**Pull request by gitstart-flanksource**
Closes #1049

### CODE REVIEW

**Comment by gitstart** [comment]
@moshloop 

We have identified certain issues with the recommended [library](https://www.npmjs.com/package/json-schema-yup-transformer) used to convert the JSON schemas to yup. One of the drawbacks is that it does not validate incorrect property names; it solely focuses on validating required fields, which may not be efficient enough. Additionally, it tends to throw errors if an unrequired object property, that has a required child property, is not provided. However, we are currently exploring alternative libraries that may offer better functionality. WDYT?

Screenshot:
image

**Comment by gitstart-flanksource** [comment]
@moshloop 

We have identified certain issues with the recommended [library](https://www.npmjs.com/package/json-schema-yup-transformer) used to convert the JSON schemas to yup. 

1. it does not validate incorrect property names; it solely focuses on validating required fields, which may not be efficient enough. 
2. it tends throws an error if an unrequired object property, that has a required child property, is not provided. 

So, we are currently exploring alternative libraries that may offer better functionality. WDYT?

File: src/components/SchemaResourcePage/SchemaResourceEdit.tsx [hunk 1]
```diff
@@ -143,7 +161,12 @@ export function SchemaResourceEdit({
     });
   }, [register, formFields, watch]);
 
-  const doSubmit = (props: any) => {
+  const doSubmit = async (props: any) => {
+    const specErrorMessage = await schemaSpecValidator(props?.["spec"]);
+    if (specErrorMessage) {
+      console.error(specErrorMessage);
```
Comment (moshloop) [inline src/components/SchemaResourcePage/SchemaResourceEdit.tsx:167]: These error messages should be shown on the form / modal

Comment (gitstart-flanksource) [reply src/components/SchemaResourcePage/SchemaResourceEdit.tsx]: Done @moshloop

**Pull request closed without merging**